src/gwt/src/org/rstudio/core/client/command/ShortcutManager.java

```diff
@@ -528,10 +529,11 @@ else if (binding.getId() == "closeSourceDoc")
             }
             
             event.stopPropagation();
-            binding.execute();
+            if (!reportShortcutBinding(binding.getId()))
+               binding.execute();
             return true;
          }
-         
+         reportShortcutBinding("Shortcut not bound"); 
          if (map.isPrefix(keyBuffer_))
```

Comment:
I think that it should be clearer whether we're at leaf node in the shortcut tree (i.e. if you have a shortcut that uses C-X as a prefix, C-X should show that it's putting us in pending mode rather than reporting C-X as totally unbound)

Reply:
Good idea, now implemented.